dali/server/daldap.cpp

```diff
@@ -144,24 +146,50 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
         Owned<ISecUser> user = ldapsecurity->createUser(username);
         user->credentials().setPassword(password);
 
-        //Check user's digital signature, if present
         bool authenticated = false;
-        if (!isEmptyString(udesc->querySignature()))
+
+        //Check that the digital signature provided by the caller (signature of
+        //caller's "username;timeStamp") matches what we expect it to be
+        if (!isEmptyString(udesc->queryUserTimeStampSignature()))
         {
             if (nullptr == pDSM)
                 pDSM = createDigitalSignatureManagerInstanceFromEnv();
             if (pDSM && pDSM->isDigiVerifierConfigured())
             {
-                StringBuffer b64Signature(udesc->querySignature());
-                if (!pDSM->digiVerify(username, b64Signature))//digital signature valid?
+                StringBuffer requestTimestamp;
+                udesc->queryUTCTimeStamp().getString(requestTimestamp, false);//extract timestamp string from Dali request
+
+                CDateTime now;
+                now.setNow();
+                if (now.compare(udesc->queryUTCTimeStamp()) < 0)//timestamp from the future?
```

Comment:
will be past if 'old' 3.14 afaics, so will deny all old <= 3.14 clients?

Reply:
Older clients will still be sending the legacy style username and password, and this new code will allow it. In this new world, a digisign aware Dali (public/private key files specified in environment) will first verify a signature if it was provided.  If the vierfication fails, the request fails, and if it passee we bypass LDAP authentication.  If no signature is provided in the request, we attempt a LDAP username/pwd authentication.